lib/inspec/file_provider.rb

```diff
@@ -7,6 +7,7 @@
 module Inspec
   class FileProvider
     def self.for_path(path)
+      raise "Profile or dependency path not resolved." if path.nil?
```

Comment:
why do we have to raise this error ?

Reply:
Because if the path is nil, it throws a stack trace with a ruby error. So this is just a nil check handling. Something must be wrong because of which the dependency is not resolved and hence the path is nil.

Reply:
this method has a bulk of if-else which eventually has a else block which should actually be run when no path is nil. What error are you facing ? may be it is not even related to this method..